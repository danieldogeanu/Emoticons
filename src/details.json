{
	"labels": [
		{
			"class": "face",
			"name": "Face"
		},
		{
			"class": "name",
			"name": "Emoticon Name"
		},
		{
			"class": "copy",
			"name": "Get"
		}
	],
	"author": {
		"name": "Daniel Dogeanu",
		"url": "https://danieldogeanu.com"
	},
	"social": [
		{
			"name": "Twitter",
			"url": "https://twitter.com/danieldogeanu"
		},
		{
			"name": "GitHub",
			"url": "https://github.com/danieldogeanu"
		},
		{
			"name": "LinkedIn",
			"url": "https://www.linkedin.com/in/danieldogeanu/"
		},
		{
			"name": "Behance",
			"url": "https://behance.net/danieldogeanu"
		}
	],
	"shortcuts": [
		{
			"key": "s",
			"description": "Selects Search Input"
		},
		{
			"key": "esc",
			"description": "Clears Search Input / Exits Menu"
		},
		{
			"key": "k",
			"description": "Opens Keyboard Shortcuts Screen"
		},
		{
			"key": "t",
			"description": "Scrolls Back Up"
		}
	]
}
