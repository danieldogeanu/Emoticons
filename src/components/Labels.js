import React from 'react';
import Label from '../elements/Label';
import details from '../details.json';

const Labels = () => (
	<div className="Labels">
		{details.labels.map((label) => (
			<Label key={label.class} label={label} />
		))}
	</div>
);

export default Labels;
