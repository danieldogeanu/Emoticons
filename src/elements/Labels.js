import React from 'react';
import Label from './Label';
import details from '../details.json';

const Labels = () => (
	<div className="Labels" data-testid="Labels">
		{details.labels.map((label) => (
			<Label key={label.class} label={label} />
		))}
	</div>
);

export default Labels;
