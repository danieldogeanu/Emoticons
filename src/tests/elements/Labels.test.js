import React from 'react';
import {render} from '@testing-library/react';
import Labels from '../../elements/Labels';
import {labels} from '../../details.json';


describe('Labels Element', () => {

	it('renders labels element properly', () => {
		const {getByTestId} = render(<Labels />);
		const renderedLabels = getByTestId('Labels');

		expect(renderedLabels).toBeInTheDocument();
		expect(renderedLabels).toHaveClass('Labels');
		expect(renderedLabels).toMatchSnapshot();
	});

	it('renders all list columns labels properly', () => {
		const {getByText} = render(<Labels />);

		labels.forEach(label => {
			const renderedLabel = getByText(label.name);

			expect(renderedLabel).toBeInTheDocument();
			expect(renderedLabel).toHaveClass('Label', label.class);
			expect(renderedLabel).toHaveTextContent(label.name);
		});
	});

});
