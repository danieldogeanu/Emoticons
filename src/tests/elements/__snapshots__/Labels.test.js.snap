// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labels Element renders labels element properly 1`] = `
<div
  class="Labels"
  data-testid="Labels"
>
  <span
    class="Label face"
  >
    Face
  </span>
  <span
    class="Label name"
  >
    Emoticon Name
  </span>
  <span
    class="Label copy"
  >
    Get
  </span>
</div>
`;
